sshd-core/src/main/java/org/apache/sshd/common/session/helpers/AbstractSession.java

```diff
@@ -712,6 +762,17 @@ protected boolean handleServiceRequest(String serviceName, Buffer buffer) throws
         return true;
     }
 
+    private boolean validateServiceKexState(KexState state) {
```

Comment:
Let's make this *protected* in the spirit of letting our users override whatever they like.

Reply:
Done.